main.go

```diff
@@ -227,6 +227,28 @@ func handleRequest(defaultConfig *libhoney.Config, w http.ResponseWriter, req *h
 		}
 		return
 	}
+	// Load potential custom query parameters
+	// e.g. API key, dataset name, and honeycomb host
+	cfg := &libhoney.Config{}
+	query := req.URL.Query()
+	key := query.Get("key")
+	if key != "" {
+		cfg.APIKey = key
+	} else {
+		cfg.APIKey = defaultConfig.APIKey
```

Comment:
This code can be simplified a little by setting all the variables to do defaults and then if they are present overload them.

```go
cfg.APIKey = defaultConfig.APIKey
cfg.Dataset = defaultConfig.Dataset
...
if dataset != "" {
		cfg.Dataset = dataset
}
```

No else statements

Reply:
That is much cleaner, thanks!